src/v/cloud_topics/level_one/compaction/meta.h

```diff
@@ -94,4 +107,11 @@ struct object_output_t {
     std::unique_ptr<staging_file> staging_file;
 };
 
+using cmp_t
+  = std::function<bool(const log_compaction_meta*, const log_compaction_meta*)>;
+using log_compaction_queue = std::priority_queue<
+  log_compaction_meta*,
+  chunked_vector<log_compaction_meta*>,
```

Comment:
consider using `ss::lw_shared_ptr`, this will prevent issues from someone taking a pointer to a member in `chunked_vector<log_compaction_meta>` or other containers that can move values around, as well as other lifetime guarantees. The main reason I bring this up is that the queue type being defined here makes me thing it's a non local abstraction, which means reasoning about the pointer lifetimes might be difficult.

Reply:
Oh use foreign_ptr then